xbmc/dbwrappers/sqlitedataset.cpp

```diff
@@ -367,28 +374,28 @@ int SqliteDatabase::connect(bool create)
   return DB_CONNECTION_NONE;
 }
 
-bool SqliteDatabase::exists(void)
+bool SqliteDatabase::exists()
 {
   bool bRet = false;
   if (!active)
     return bRet;
   result_set res;
-  char sqlcmd[512];
 
   // performing a select all on the sqlite_master will return rows if there are tables
   // defined indicating it's not empty and therefore must "exist".
-  snprintf(sqlcmd, sizeof(sqlcmd), "SELECT * FROM sqlite_master");
-  if ((last_err = sqlite3_exec(getHandle(), sqlcmd, &callback, &res, NULL)) == SQLITE_OK)
+  const std::string sqlcmd{"SELECT * FROM sqlite_master"};
```

Comment:
nitpick, seems wasteful to build a string only for the C sqlite API.
`const char sqlCmd[] = "SELECT * FROM sqlite_master";` would do?

Reply:
okay. will change that.

Reply:
done